src/Numerics/Interpolation/LinearSpline.cs

```diff
@@ -130,6 +130,24 @@ public static LinearSpline Interpolate(IEnumerable<double> x, IEnumerable<double
         /// <returns>Interpolated value x(t).</returns>
         public double Interpolate(double t)
         {
+            return Interpolate(t, true);
+        }
+
+        /// <summary>
+        /// Interpolate at point t.
+        /// </summary>
+        /// <param name="t">Point t to interpolate at.</param>
+        /// <param name="allowExtrapolate">set to fals if extrapolate should not be allowed</param>
+        /// <returns>Interpolated value x(t).</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when t is outside range and would result in Extrapolation!</exception>
+        public double Interpolate(double t, bool allowExtrapolate)
```

Comment:
I am not a big fan of ```bool``` variables in public APIs which act like toggle switches. I would rather create another public function, something like ```InterpolateWithPossibleExtrapolation```, which probably needs to defined on the interface, so every instance can implement it.

Reply:
Hi again. I am in favor of having a clean API, and I feel that your proposal is in the right direction.

You are right, the name I suggested should have been ``` InterpolateWithoutExtrapolation ```